src/adapter/src/coord/sequencer.rs

```diff
@@ -471,6 +472,32 @@ impl Coordinator {
         }
     }
 
+    // Utilities used by `sequence` functions
+    // ---
+    fn ensure_cluster_is_not_linked(&self, cluster_id: ClusterId) -> Result<(), AdapterError> {
+        if let Some(linked_id) = self.catalog.get_linked_object(cluster_id) {
```

Comment:
I think `get_linked_object` is not worth it's weight! Given that you then have to fetch the cluster below. Can we instead do:

```
let cluster = self.catalog.get_cluster(cluster_id);
if let Some(linked_id) = cluster.linked_object_id {
    let cluster_name = cluster.name.clone();
}
```